Test_Data_Sender.cpp

```diff
@@ -0,0 +1,23 @@
+#define CATCH_CONFIG_MAIN
+
+#include "test/catch.hpp"
+#include "Data_Sender.hpp"
+
+TEST_CASE("Check if data is read from the file")
```

Comment:
Readable and well understandable code.
Test cases can be extended to verify the print on console functionality too.